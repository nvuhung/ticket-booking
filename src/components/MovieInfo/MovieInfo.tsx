import React from 'react';
import styled from 'styled-components';

import { theme } from '../../styles/theme';
import { IMovie } from '../../models/movie';

export interface IProps {
  movie: IMovie | null;
}

const MovieInfo: React.FC<IProps> = ({ movie }) =>
  movie ? (
    <>
      <Title>{movie.name}</Title>
      <Text>{movie.description}</Text>
    </>
  ) : null;

const Title = styled.h3`
  text-align: center;
  color: ${theme.colors.white};
`;

const Text = styled.p`
  text-align: center;
  color: ${theme.colors.white};
`;

export default React.memo(MovieInfo);
