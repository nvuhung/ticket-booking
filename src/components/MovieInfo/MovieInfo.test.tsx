import React from 'react';
import 'jest-styled-components';

import MovieInfo, { IProps } from './MovieInfo';
import { mount, shallow } from '../../test/enzyme';
import { seats } from '../../test/mocks';

describe('MovieInfo', () => {
  let props: IProps = (null as any) as IProps;
  let mountedComp: any;

  const wrapper = (bShallow: boolean) => {
    if (!mountedComp) {
      mountedComp = bShallow ? shallow(<MovieInfo {...props} />) : mount(<MovieInfo {...props} />);
    }
    return mountedComp;
  };

  beforeEach(() => {
    props = {
      movie: null
    };
  });

  it('renders without crashing', () => {
    const comp = wrapper(false);
    expect(comp).toMatchSnapshot();
  });

  describe('when movie is null', () => {
    beforeEach(() => {
      (props.movie = null), (mountedComp = undefined);
    });

    it('should not render', () => {
      const comp = wrapper(false);
      expect(comp.find('p').length).toEqual(0);
    });
  });

  describe('when movie has value', () => {
    beforeEach(() => {
      (props.movie = seats.movie), (mountedComp = undefined);
    });

    it('should render contents', () => {
      const comp = wrapper(false);
      expect(comp.find('h3').length).toEqual(1);
      expect(comp.find('p').length).toEqual(1);
    });

    it('should match data', () => {
      const comp = wrapper(false);
      const title = comp.find('h3');
      const description = comp.find('p');
      expect(title.text()).toEqual('Spider-man: Người nhện xa nhà');
      expect(description.text()).toEqual('C13 | 2D Vietnam sub');
    });
  });
});
