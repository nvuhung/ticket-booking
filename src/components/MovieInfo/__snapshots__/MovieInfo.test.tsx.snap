// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo renders without crashing 1`] = `
<Memo(MovieInfo)
  movie={null}
/>
`;
